### DISCUSSION TITLE
How to modify the device resources Profile  in deivce service via consul?

### DISCUSSION THREAD

**Discussion by zenosp** [edgexfoundry/community#9] [Q&A]
Hi,

I can't find the entry to modify the device resource file. Consul only show the variable ProfilesDir, so I can not fiind the content of ProfilesDir below.

![图片](https://user-images.githubusercontent.com/14228338/189358876-fef58ab7-dc1d-40a9-81f8-ed6ac384d970.png)

**Comment by farshidtz** [comment]
The device and profile configurations are maintained in the Core Metadata service, not in Config Provider (Consul). The ProfilesDir path is the path to the profile files that are bootstrapped (and uploaded to Core Metadata if those profiles don't already exist) during the service startup.

Please refer to:
https://docs.edgexfoundry.org/2.2/microservices/core/metadata/Ch-Metadata/
https://docs.edgexfoundry.org/2.2/api/core/Ch-APICoreMetadata/

**Comment by zenosp** [comment]
Hi farshidtz,

Thanks for your reply. 

I read the code of device-sdk-c, it is consistent with what you said, the Device Service will upload the local device profile and resources information to the Core-Meta Data Service when it started and cannot get device profile.

However, in the Core-Meta Data Service, I cannot modify the device profile through consul.

According to my understanding, device profile is a part of device service, which should be exposed to consul for modification.

Best Regards.

**Reply by farshidtz** [reply]
The device and device profile resources are NOT kept in Consul. They are stored inside the Core Metadata itself. You can use the [REST API of Core Metadata](https://docs.edgexfoundry.org/2.2/api/core/Ch-APICoreMetadata/) or the [EdgeX CLI](https://docs.edgexfoundry.org/2.2/getting-started/tools/Ch-CommandLineInterface/) to query them.
